docs/api.md

```diff
@@ -2,55 +2,81 @@
 
 ## Async API Overview
 
-The `AsyncHTTPTransport` and `AsyncByteStream` classes provide the base
-interface which transport classes need to implement.
+### Base async interfaces
 
-::: httpcore.AsyncHTTPTransport
-    :docstring:
-    :members: arequest aclose
+These classes provide the base interface which transport classes need to implement.
 
-::: httpcore.AsyncByteStream
-    :docstring:
-    :members: __aiter__ aclose
+:::{eval-rst}
+.. autoclass:: httpcore.AsyncHTTPTransport
+    :members: arequest, aclose
 
-The `AsyncConnectionPool` class is a concrete implementation of `AsyncHTTPTransport`.
+.. autoclass:: httpcore.AsyncByteStream
+    :members: __aiter__, aclose
+:::
 
-::: httpcore.AsyncConnectionPool
-    :docstring:
+### Async connection pool
 
+:::{eval-rst}
+.. autoclass:: httpcore.AsyncConnectionPool
+    :show-inheritance:
+:::
 
-The `PlainByteStream` and `AsyncIteratorByteStream` classes are concrete implementations of `AsyncByteStream`.
+### Async proxy
 
-::: httpcore.PlainByteStream
-    :docstring:
+:::{eval-rst}
+.. autoclass:: httpcore.AsyncHTTPProxy
+    :show-inheritance:
+:::
 
-::: httpcore.AsyncIteratorByteStream
-    :docstring:
+### Async byte streams
 
----
+These classes are concrete implementations of [`AsyncByteStream`](httpcore.AsyncByteStream).
+
+:::{eval-rst}
+.. autoclass:: httpcore.PlainByteStream
+    :show-inheritance:
+
+.. autoclass:: httpcore.AsyncIteratorByteStream
+    :show-inheritance:
+:::
 
 ## Sync API Overview
 
-The `SyncHTTPTransport` and `SyncByteStream` classes provide the base
-interface which transport classes need to implement.
+### Base sync interfaces
+
+These classes provide the base interface which transport classes need to implement.
+
+:::{eval-rst}
+.. autoclass:: httpcore.SyncHTTPTransport
+    :members: request, close
+
+.. autoclass:: httpcore.SyncByteStream
+    :members: __iter__, close
+:::
+
+### Sync connection pool
 
-::: httpcore.SyncHTTPTransport
-    :docstring:
-    :members: request close
+:::{eval-rst}
+.. autoclass:: httpcore.SyncConnectionPool
+    :show-inheritance:
+:::
 
-::: httpcore.SyncByteStream
-    :docstring:
-    :members: __iter__ close
+### Sync proxy
 
-The `SyncConnectionPool` class is a concrete implementation of `SyncHTTPTransport`.
+:::{eval-rst}
+.. autoclass:: httpcore.SyncHTTPProxy
+    :show-inheritance:
+:::
 
-::: httpcore.SyncConnectionPool
-    :docstring:
+### Sync byte streams
 
-The `PlainByteStream` and `IteratorByteStream` classes are concrete implementations of `SyncByteStream`.
+These classes are concrete implementations of [`SyncByteStream`](httpcore.SyncByteStream).
 
-::: httpcore.PlainByteStream
-    :docstring:
+:::{eval-rst}
+.. autoclass:: httpcore.PlainByteStream
+    :show-inheritance:
+    :noindex:
```

Comment:
Curious - what is `:noindex:` here, and why on this class in particular?

Reply:
We show this class two times, once for the sync API, once for the async API. I figured it's fine, except showing them twice would make Sphinx index them twice, which raised a warning hinting to add `:noindex:`.